### ISSUE TITLE
[FEATURE] - Jump / Toggle next bookmark

### ISSUE THREAD

**Issue by zardoy** [alefragnani/vscode-numbered-bookmarks#152]
Hi! Idea is great. I'm currently not really satisfied with the implementation here. For example I can't toggle next available bookmark. For example if bookmarks 1 and 2 are already here in current editor I'd like to toggle 3rd bookmark. If all possible numbered bookmarks are already here, I'd like to see a message about that. I could imagine about workaround if this extension used `setContext` under the hood so for example I had `numberedBookmarks.bookmark1Toggled` when context to check in my command.

Also why don't you bring ability to go to next / previous bookmark from your Bookmarks extension? It can be useful in some scenarios, but of course this is not that important.

Sorry if I misunderstood something.

**Comment by alefragnani** [comment]
Hi @zardoy ,

The extension was created to replicate the Delphi IDE experience, which was my main IDE/Language for a long time. In fact, Delphi was the only IDE/Editor that I've seen to have native Numbered Bookmarks feature, up until IntelliJ. In the end, that's the main reason why this extension works this way.

In fact, it's a bit odd to me the extension to create the _next number_ to me, because I always understood the numbered bookmark as a _specific mnemonic that I decided to use_. I mean, I always define the number that I want for a specific bookmark based on its _importance_. When I want something automatic, without limits, I use the **Bookmarks** extension instead.

I have some questions about how it could work within the extension:

**Comment by zardoy** [comment]
Sorry for delay. I'm just not really motivated of this issue anymore, but I had a usecase where I needed to refactor a giant file and I was marking locations to refactor with mentioned standard bookmarks extension. But it was easy to forget what exactly I wanted to do. With the numbers I can associate my thoughts. Also I was doing tasks from last to first. Numbers + this automatic feature would be ideal for this in future. In other words the main problem in standard bookmarks extension is that all bookmarks are the same and I can't track the order in which they were placed. Sorry if I didn't make it clear, feel free to ask :)

**Comment by alefragnani** [comment]
Yup, it is clear to me. That's why I asked how that two scenarios would work with such behavior.

In fact, another way to think about your scenario is a _stack behavior_ for bookmarks, where you can navigate _back and forth_ in history.

**Comment by ckx** [comment]
Thanks for the great extension @alefragnani. I came here to request this feature. Here is my use case and some general appreciation for you:

I use Numbered Bookmarks primarily as "temporary bookmarks", typically with a lifetime of a couple days. It has been a life saver for me, allowing me to efficiently jump around a very large C codebase where, it isn't unusual for files to be 10,000-30,000 lines long. To make matters worse, things are quite disorganized with most systems spanning multiple files and having a heavy reliance on globals, etc. 

I use the `List from All files` feature to navigate the codebase more than any other tool. This feature has transformed my nightmare scenario into something much more manageable, I do not let out a heavy sigh every time I have to context switch to another area of the code any longer.

Anyway, I often lose track of which bookmarks I've already placed in any given file, and when I accidentally move an existing bookmark, it can be disruptive to my workflow. Pressing my hotkey to "Clear bookmarks" is also scary in this scenario, so I tend to leave some bookmarks around until I'm done with my work on any given system, making it even harder to remember which I've used.

All this to say, stack behavior for bookmarks would be almost life-changing for me, as it'd allow me to just press a hotkey like "Use next free bookmark" and know my other bookmarks are fine. It's something that crosses my mind daily. Is this feature currently on your radar?
